dask/dataframe/groupby.py

```diff
@@ -514,7 +514,11 @@ def _non_agg_chunk(df, *by, key, dropna=None, observed=None, **kwargs):
             # If we found any categoricals, append unobserved values to the end of the
             # frame.
             new_cats = full_index[~full_index.isin(result.index)]
-            empty = pd.DataFrame(pd.NA, index=new_cats, columns=result.columns)
+            empty_data = {
+                col: pd.Series([None], name=col, dtype=result[col].dtype)
```

Comment:
Can we specify the Series through 
```
pd.Series(index=new_cats, dtype=result[col].dtype)
```

and remove the index in the Frame constructor? This is more efficient, because it won't trigger a reindexing operation internally

Reply:
I applied the suggestion, thanks for the tip!